src/App.jsx

```diff
@@ -1,13 +1,92 @@
-import "./App.css";
-import TodoForm from "./Components/TodoForm";
+import "./App.css"
+import TodoForm from "./Components/TodoForm"
+import { useState, useEffect } from "react"
+import { nanoid } from "nanoid"
+import { parseISO, isToday, isThisWeek } from "date-fns"
 
 function App() {
-  return (
-    <>
-      <h1 className="text-3xl font-bold underline">Hello world!</h1>
-      <TodoForm />
-    </>
-  );
+    const [inputValue, setInputValue] = useState("")
+    const [dateValue, setDateValue] = useState(null)
+    const [todos, setTodos] = useState([
+        { id: nanoid(), name: "Buy coffee" },
+        { id: nanoid(), name: "Water plants" },
+    ])
+    const [filteredTodos, setFilteredTodos] = useState([])
+    const [currentFilter, setCurrentFilter] = useState("inbox")
+
+    //filter todos and update dom on filter & todo change//
+    useEffect(() => {
+        if (currentFilter === "inbox") {
+            setFilteredTodos([...todos])
+        }
+        if (currentFilter === "today") {
+            setFilteredTodos(
+                todos.filter(todo => {
+                    return isToday(parseISO(todo.date))
+                })
+            )
+        }
+        if (currentFilter === "thisWeek") {
+            setFilteredTodos(
+                todos.filter(todo => {
+                    return isThisWeek(parseISO(todo.date))
+                })
+            )
+        }
+    }, [todos, currentFilter])
+
+    return (
+        <>
+            <h1 className="text-3xl font-bold underline">Hello world!</h1>
+            <header>
+                <nav>
+                    <button
+                        onClick={() => {
+                            setCurrentFilter("inbox")
+                        }}
+                    >
+                        Inbox
+                    </button>
+                    <button
+                        onClick={() => {
+                            setCurrentFilter("today")
+                        }}
+                    >
+                        Today
+                    </button>
+                    <button
+                        onClick={() => {
+                            setCurrentFilter("thisWeek")
+                        }}
+                    >
+                        This Week
+                    </button>
+                </nav>
+            </header>
+            <TodoForm
```

Comment:
There is no need to pass these props into TodoForm. And best practice tip is not to pass setter functions as props :)